whois-api/src/main/java/net/ripe/db/whois/api/rdap/RdapRequestValidator.java

```diff
@@ -8,59 +8,64 @@
 import net.ripe.db.whois.common.rpsl.attrs.Domain;
 import net.ripe.db.whois.update.domain.ReservedAutnum;
 import static net.ripe.db.whois.common.rpsl.ObjectType.MNTNER;
+import static org.apache.commons.lang.StringUtils.isEmpty;
 import org.springframework.beans.factory.annotation.Autowired;
+import javax.ws.rs.BadRequestException;
+import javax.ws.rs.NotFoundException;
 import org.springframework.stereotype.Component;
 
 import static net.ripe.db.whois.common.rpsl.ObjectType.ORGANISATION;
 
 @Component
 public class RdapRequestValidator {
 
-    private final RdapExceptionMapper rdapExceptionMapper;
     private final ReservedAutnum reservedAutnum;
 
     @Autowired
-    public RdapRequestValidator(final RdapExceptionMapper rdapExceptionMapper, final ReservedAutnum reservedAutnum) {
-        this.rdapExceptionMapper = rdapExceptionMapper;
+    public RdapRequestValidator(final ReservedAutnum reservedAutnum) {
         this.reservedAutnum = reservedAutnum;
     }
 
     public void validateDomain(final String key) {
+        if (isEmpty(key)) {
+            throw new BadRequestException("empty lookup term");
+        }
+
         try {
             Domain.parse(key);
         } catch (AttributeParseException e) {
-            throw rdapExceptionMapper.notFound("RIPE NCC does not support forward domain queries.");
+            throw new NotFoundException("RIPE NCC does not support forward domain queries.");
```

Comment:
were the badRequest and notFound methods redundant? I.e. is the response any different?

Reply:
As discussed, these exceptions are handled in RdapExceptionMapper, and correctly wrapped in an RDAP-specific JSON response body.